Feature/codecheck submission fields

subhanLabs opened the pull request:
- Add opt-in checkbox and metadata fields to submission wizard
- Create Vue components for repository and manifest file inputs
- Implement auto-save and review panel display
- Add schema fields for code/data repository information
- Include styling and internationalization support
---

## Related User Stories

- A-ME1 (related): As an author, I want to enter CODECHECK-related metadata (code URL, data URL, expected outputs) only once...

_User story references from the [CHECK-PUB User Stories document](https://codecheck.org.uk/pub/)._

subhanLabs committed:
refactor: Convert CODECHECK submission fields from vanilla JS to Vue.js

subhanLabs committed:
Merge branch 'main' into feature/codecheck-submission-fields

subhanLabs committed:
Add CODECHECK submission wizard integration

- Add opt-in checkbox and metadata fields to submission wizard
- Create Vue components for repository and manifest file inputs
- Implement auto-save and review panel display
- Add schema fields for code/data repository information
- Include styling and internationalization support

nuest reviewed (changes requested):
Found just small details.

Subhanaliweb commented on resources/js/main.js:
```diff
@@ -0,0 +1,345 @@
+    }
+  }
+
+  escapeHtml(text) {
```
The function name is correct, it does escape HTML, it's used to prevent XSS attacks when displaying user-provided data in the review panel.

How it escapes:
1. `div.textContent = text` - The browser automatically escapes HTML special characters when setting textContent
2. `div.innerHTML` - Returns the escaped version

Example:
- Input: `<script>alert('XSS')</script>`
- Setting textContent causes browser to escape: `< → &lt;` and `> → &gt;`
- Reading innerHTML returns: `&lt;script&gt;alert('XSS')&lt;/script&gt;`
- This escaped string is safe to display - browsers show it as text, not executable code

This is a standard JavaScript pattern for HTML escaping without external libraries.RetryKA

subhanLabs committed:
Use locale replacement variable for opt-in checkbox link (RTL support
- Replace string concatenation with {} placeholder
- Enable proper link positioning for RTL languages

dxL1nus commented:
@nuest Did we really still ned this PR? We now have the fields again in the Submission Form, although we clarified a couple meetings ago, that we don't want that there. I am just curious, because when my PR gets merged, I want to know if I should keep that in when I merge main to my branch or directly take these submission field out again.

nuest commented:
We have another PR open for the metadata form: https://github.com/codecheckers/ojs-codecheck/pull/59

@Subhanaliweb Can you please double-check that the fields required during submission are really only the ones that an author can fill out? Please update the submission form accordingly (in the other PR?).

@dxL1nus Thanks for spotting this.

This reminds me on two things: We need to make time for actual reproductions, and we need to use smaller tasks with mock-ups.
